Adds more customization options to the Chip widget

paolosoares opened the pull request:
Includes:
- Custom Tooltip message for the delete button;
- Custom background color for the chip
- Custom delete icon color
- Custom label text style

paolosoares commented:
Updated e-mail to match CLA.

Hixie commented:
This is great, can you add some tests for these too?

cc @abarth who may find some wry amusement at our continued experiences with this widget.

paolosoares commented:
Expanded the chip tests to verify the delete labels.

paolosoares committed:
Adds more customization options to the Chip widget

Includes:
- Custom Tooltip message for the delete button;
- Custom background color for the chip
- Custom delete icon color
- Custom label text style

Hixie commented on packages/flutter/lib/src/material/chip.dart:
```diff
@@ -74,6 +78,18 @@ class Chip extends StatelessWidget {
+  /// Background color for the widget.
+  final Color backgroundColor;
+
+  /// Color for the "Delete" icon.
```
these docs should reference onDeleted and say that it's ignored if that's null.

Hixie replied:
also, no need to put the word Delete in quotes since it's not in quotes in the other docs.

Hixie commented on packages/flutter/lib/src/material/chip.dart:
```diff
@@ -74,6 +78,18 @@ class Chip extends StatelessWidget {
+  /// The style to be used for the label text.
+  final TextStyle labelStyle;
+
+  /// Message for the chip delete button's tooltip.
```
these docs should reference onDeleted and say that it's ignored if that's null.
￼

Hixie commented on packages/flutter/lib/src/material/chip.dart:
```diff
@@ -74,6 +78,18 @@ class Chip extends StatelessWidget {
+  /// Message for the chip delete button's tooltip.
+  final String deleteButtonTooltipMessage;
+
+  /// Background color for the widget.
```
These docs aren't detailed enough. We can guess that a property called "backgroundColor" on a widget gives the background color of the widget. :-)

What it should say is where the color appears, what the default is, etc.

Hixie commented:
Travis caught a few places in the test that were missing type annotations.

Hixie commented:
Patch looks good overall, just a few fixes before we can check it in!

Hixie commented:
Thanks for your contribution!

paolosoares committed:
Adds missing type annotations to tests and improves documentation.

Hixie commented on packages/flutter/lib/src/material/chip.dart:
```diff
@@ -74,6 +78,29 @@ class Chip extends StatelessWidget {
 
+  /// The style to be applied to the chip's label.
+  ///
+  /// This only has effect on widgets that use a [TextStyle], such as [Text].
```
A better way to phrase this would be "`...that respect the [DefaultTextStyle], such as [Text]`". I mean, `RichText` uses `TextStyle` too, but it won't affect that.

Hixie commented:
This is great!
![LGTM](http://junkyard.damowmow.com/579)

Hixie commented:
Let me know if you make the slight change suggested to the style docs, and I'll land it for you!

paolosoares committed:
Tweaks labelStyle field documentation

paolosoares commented:
I've made the requested changes in the docs.

Hixie commented:
Fantastic! Thanks!
